pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -74,8 +74,12 @@ func convertService(svc v1.Service, domainSuffix string) *model.Service {
 
 	if svc.Spec.Type == v1.ServiceTypeExternalName && svc.Spec.ExternalName != "" {
 		external = svc.Spec.ExternalName
-		resolution = model.Passthrough
-		meshExternal = true
+		if strings.HasSuffix(external, "."+domainSuffix) {
+			resolution = model.DNSLB
```

Comment:
add a comment here as to why you are doing this (that you can back an exterbal service with in cluster service)

Reply:
Have a doubt: is this a reasonable way to use  ExternalName service? How many users would use it like this? Is it worth the complexity?

 ExternalName service is meant to expose external service(not to mask
internal service) to cluster.

Reply:
Thats a good point.. Until this PR, I did not even know that its possible to use external name services in this way.

Reply:
Look, `ExternalName` is obviously a victim of bad naming here. But it's already in the k8s API as is. And it works, without istio.

I want my k8s services to be usable, both without and **with** istio. 

My use case: I have a facade service and two internal services (one of them in a different namespace), and I need to **switch** the facade between them.